Client/Frontend/Library/BookmarksPanel.swift

```diff
@@ -117,13 +118,22 @@ class BookmarksPanel: SiteTableViewController, LibraryPanel {
                 // TODO: Handle error case?
                 self.bookmarkFolder = nil
                 self.bookmarkNodes = []
-                self.tableView.reloadData()
+                self.recentBookmarks = []
                 return
             }
 
             self.bookmarkFolder = folder
             self.bookmarkNodes = folder.children ?? []
-            self.tableView.reloadData()
+
+            if folder.guid == BookmarkRoots.RootGUID {
+                self.profile.places.getRecentBookmarks().uponQueue(.main) { result in
```

Comment:
Move this into `loadData()` so that it runs concurrently with `getBookmarksTree()`. It also will run it automatically on `viewWillAppear()` so it will refresh itself each time the view is shown.